wgpu-types/src/lib.rs

```diff
@@ -702,37 +718,51 @@ impl DownlevelCapabilities {
 }
 
 bitflags::bitflags! {
-    /// Binary flags listing various ways the underlying platform does not conform to the WebGPU standard.
+    /// Binary flags listing features that may or may not be present on downlevel adapters.
+    ///
+    /// Flags that are **not** present for a downlevel adapter or device usually indicates
+    /// non-compliance with the WebGPU specification, but not always.
+    ///
+    /// You can check whether a set of flags is compliant by by comparing to
+    /// [`DownlevelFlags::compliant()`]:
+    ///
+    /// ```
+    /// # let my_flags = DownlevelFlags::all();
+    /// # use wgpu_types::DownlevelFlags;
+    /// assert_eq!(my_flags & DownlevelFlags::compliant(), DownlevelFlags::compliant());
+    /// ```
     pub struct DownlevelFlags: u32 {
         /// The device supports compiling and using compute shaders.
-        const COMPUTE_SHADERS = 0x0000_0001;
+        const COMPUTE_SHADERS = 1 << 0;
```

Comment:
Let's not change more than necessary in this PR. It shouldn't affect wgpu-types if possible, we can discuss these changes separately.
Also, it would be great to land smaller patches gradually. This PR is growing to be big, and already colliding with other incoming changes like #1717

Reply:
Sorry, I just realized #1717 is by you, and it's doing precisely what I'm asking about - moving things out of the giant PR and landing them separately. Thank you for doing this!

Reply:
No problem. :) I do have a habit of shoving a bunch of stuff into one PR/commit.

There probably is still quite a bit of splitting up that I could do so I'll be trying to clean this up a bit and opening separate PRs when possible.